### PR TITLE
Add support for a lot of quaternion operations, quaternion tests and tester script

### LINKED ISSUE #63: Write unit tests for quaternion

### LINKED ISSUE #107: Add a script for easier manual testing
Ideally, it would simplify the whole testing process to just doing something like `sbcl --load tester.lisp` and `(lbge-unit-tests:run)`

### PR DESCRIPTION
**Pull request by fami-com**
Closes #107, partially closes #63 (no tests for `to-euler` and `from-euler`)

### CODE REVIEW

File: src/math/quaternion.lisp [hunk 1]
```diff
@@ -9,6 +9,103 @@
...
+         (z (quaternion-z q))
+         (xsgn (get-sign x))
+         (ysgn (get-sign y))
+         (zsgn (get-sign z)))
+      (format out "~a ~a ~ai ~a ~aj ~a ~ak" w xsgn (abs x) ysgn (abs y) zsgn (abs z))))
+
+(defmacro define-quaternion-op (name op)
+  `(defmethod ,name ((q1 quaternion) (q2 quaternion))
+    (make-quaternion
+      :x (funcall ,op (quaternion-x q1) (quaternion-x q2))
```
Comment (alex-eg) [inline src/math/quaternion.lisp]: You can omit `funcall` here.
Just `(,op (quaternion-x q1) (quaternion-x q2))`

File: src/math/quaternion.lisp [hunk 2]
```diff
@@ -9,6 +9,103 @@
...
+
+(defmethod add ((value real) (q quaternion))
+  (make-quaternion
+    :w (+ (quaternion-w q) value)
+    :x (quaternion-x q)
+    :y (quaternion-y q)
+    :z (quaternion-z q)))
+
+(defmethod add ((q quaternion) (value real))
+  (make-quaternion
```
Comment (alex-eg) [inline src/math/quaternion.lisp]: You can call add with swapped arguments here.
Or extract the common part to a non-exported macro which use in these methods.

File: src/math/vector.lisp [hunk 3]
```diff
@@ -136,13 +122,13 @@
...
-  `(defmethod ,name ((vector1 ,result-type) vector2)
+  `(defmethod ,name ((vector1 ,result-type) (vector2 ,result-type))
      (make-instance ',result-type :in-vec
                     (map 'vector ,map-op
                          (in-vec vector1)
                          (in-vec vector2)))))
 
-(define-vec-op add float2 #'+)
+(define-vec-op add floatn #'+)
 (define-vec-op add float3 #'+)
```
Comment (alex-eg) [inline src/math/vector.lisp]: Do we still need these methods?

File: tester.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+RESP=$(echo '(if (member :quicklisp *features*) 1 0)' | sbcl --noinform)
+if [[ $RESP =~ 0 ]]; then
+  echo "Quicklisp not installed. Install?[yn]"
+  read -p "Are you sure? " -n 1 -r
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    curl 'https://beta.quicklisp.org/quicklisp.lisp' | sbcl
+  fi
+fi
+
+if [ ! -L "$DESTLINK" ]; then
```
Comment (alex-eg) [inline tester.sh]: If you have `[[ ]]` on line 6, it would be natural to have same notation here.

File: tester.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
```
Comment (alex-eg) [inline tester.sh]: Please move this file to `a/unit-tests` dir and add info about it in README.org.

File: src/math/t/quaternion.lisp [hunk 6]
```diff
@@ -0,0 +1,100 @@
+(define-test add-test
```
Comment (alex-eg) [inline src/math/t/quaternion.lisp]: Oh no.
We have a problem.
There already are tests with the same name, in `vector.lisp` and `matrix.lisp`.
And according to the job log, only `vector.lisp` test has actually run.

Please rename all quaternion tests to quaternion-<whatitdoes>-test, and the issue with clashing tests names I will address ASAP in the separate issue.

File: src/math/vector.lisp (hunk 3 above)
Comment (fami-com) [reply src/math/vector.lisp]: Actually yes, because I forgot that we create an instance afterwards, so we need to know the type.

**Pull request merged**